src/pages/dashboard/gig-board-create/index.tsx

```diff
@@ -0,0 +1,124 @@
+import { useContext, useState } from 'react';
+import Image from 'next/image';
+import { useRouter } from 'next/router';
+import { parseRateAmount } from '../../../utils/web3';
+import { useChainId } from '../../../hooks/useChainId';
+import useAllowedTokens from '../../../hooks/useAllowedTokens';
+import StarterKitContext from '../../../context/starterKit';
+import Steps from '../../../components/Steps';
+import ServiceForm from '../../../components/Form/ServiceForm';
+import { formatDate } from '../../../utils/dates';
+import { renderTokenAmountFromConfig } from '../../../utils/conversion';
+
+interface ILiveServiceContent {
+  title?: string;
+  about?: string;
+  rateAmount?: string;
+  rateToken?: string;
+}
+
+const GigBoardCreator = () => {
+  const [liveServiceContent, setLiveServiceContent] = useState<ILiveServiceContent>({
+    title: 'Enter a title',
+    rateToken: '0x0000000000000000000000000000000000000000',
+    rateAmount: '0',
+  });
+
+  const chainId = useChainId();
+  const allowedTokenList = useAllowedTokens();
+  const { user } = useContext(StarterKitContext);
+  const router = useRouter();
+  const navigateToGigBoardSettings = () => router.push('/dashboard/gig-board-create/settings');
+
+  if (!user) {
+    return <Steps />;
+  }
+
+  return (
+    <div className='max-w-7xl mx-auto text-gray-200 sm:px-4 lg:px-0'>
+      <p className='flex items-center text-2xl font-medium tracking-wider mb-8 border-b w-full border-gray-700'>
+        Post your First Gig
+      </p>
+      <div className='grid grid-cols-2 gap-2'>
+        <div>
+          <ServiceForm
+            onSuccess={navigateToGigBoardSettings}
+            onChange={(event: ILiveServiceContent) => {
+              // TODO: move this to an async function expression
+              if (event.rateAmount) {
```

Comment:
Best practices and perf improvement: 
here a new instance of your function is created on every rerender, better to use useCallback